tests/e2e/vm_creation_performance_test.go

```diff
@@ -0,0 +1,228 @@
+				Cores:        1,
+				CoreFraction: "25%",
+				ModelName:    "generic-v1",
+			},
+			Memory: v1alpha2.MemorySpec{Size: "1Gi"},
+			BlockDevices: []v1alpha2.BlockDeviceSpec{
+				{
+					Type: v1alpha2.DiskDevice,
+					VirtualMachineDisk: &v1alpha2.DiskDeviceSpec{
+						Name: vmdName.Name,
+					},
+				},
+			},
+			Provisioning: &v1alpha2.Provisioning{
+				Type:     v1alpha2.ProvisioningType("UserData"),
+				UserData: vmUserData,
+			},
+		},
+	}
+	return vm
+}
+
+func VMD(namespace, name string, cvmiName v1alpha2.ClusterVirtualMachineImage) v1alpha2.VirtualMachineDisk {
+
+	vmd := v1alpha2.VirtualMachineDisk{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1alpha2.VirtualMachineDiskSpec{
+			DataSource: &v1alpha2.VMDDataSource{
+				Type: v1alpha2.DataSourceTypeClusterVirtualMachineImage,
+				ClusterVirtualMachineImage: &v1alpha2.DataSourceNamedRef{
+					Name: cvmiName.Name,
+				},
+			},
+			PersistentVolumeClaim: v1alpha2.VMDPersistentVolumeClaim{
+				Size: resource.NewQuantity(6*1024*1024*1024, resource.BinarySI),
+			},
+		},
+	}
+	return vmd
+}
+
+func createVM(client kubeclient.Client, vm v1alpha2.VirtualMachine, vmd v1alpha2.VirtualMachineDisk, namespace string) {
+	GinkgoHelper()
+	resVMD, err := client.VirtualMachineDisks(namespace).Create(context.TODO(), &vmd, metav1.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred(), "cannot create vmd - %s", &vmd.Name)
+	resVM, err := client.VirtualMachines(namespace).Create(context.TODO(), &vm, metav1.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred(), "cannot create vm - %s", &vm.Name)
+	fmt.Println("VM", resVM.Name, "with VMD", resVMD.Name, "created")
+}
+
+var _ = Describe("Performance test 20 vm creation", Label("performance"), Ordered, ContinueOnFailure, func() {
+	const (
+		vmName                         = "perf-test-vm"
+		vmdName                        = "perf-test-vmd"
+		cvmiName                       = "ubuntu-22.04"
+		vmCount                        = 26
+		overallTimeout                 = 12 * time.Minute
+		deleteGracePeriodSeconds int64 = 30
+	)
+
+	var cvmi *v1alpha2.ClusterVirtualMachineImage
+	var vmMap = make(map[string]string)
+
+	clientConfig := kubeclient.DefaultClientConfig(&pflag.FlagSet{})
+	client, err := kubeclient.GetClientFromClientConfig(clientConfig)
+	if err != nil {
+		log.Fatalf("Cannot obtain Virtualization client: %v\n", err)
+	}
+
+	AfterAll(func() {
+		By("Delete all resources")
+		vmList, err := client.VirtualMachines(conf.Namespace).List(context.TODO(), metav1.ListOptions{})
+		for _, vm := range vmList.Items {
+			err = client.VirtualMachines(conf.Namespace).Delete(context.TODO(), vm.Name, metav1.DeleteOptions{
+				GracePeriodSeconds: ptr.To(deleteGracePeriodSeconds)})
+			Expect(err).NotTo(HaveOccurred())
+			if err != nil {
+				fmt.Println(err)
+			}
+		}
+	})
+
+	Context("VM", func() {
+		cvmi, err = CVMI(client, cvmiName, "create")
+		Expect(err).NotTo(HaveOccurred(), "%s", err)
```

Comment:
```suggestion
		Expect(err).NotTo(HaveOccurred(), "should create CVMI %s", cvmiName)
```

`%s, err` is redundant: ginkgo will print err, as an argument from the Expect. Better add something not in Expect but helpful for debugging, some arguments from previous call, e.g. cvmiName.